jeos-firstboot-enroll

```diff
@@ -29,7 +32,14 @@ detect_luks2()
 
 enroll_systemd_firstboot() {
 	[ -e /usr/bin/systemd-cryptenroll ] || return 0
-	detect_luks2 || return 0
+	crypt_keyid="$(keyctl id %user:cryptenroll)"
+	[ -n "$crypt_keyid" ] || return 0
+
+	if ! detect_luks2; then
+		d --msgbox $"Error: recovery set but no encrypted disks found" 0 0
```

Comment:
I would revert the condition, first check if there is a luk2 device and silently exit, and later check for the crypt_keyid, and if it is not present show this error. It is more clear.

Reply:
My thinking was that checking for the keyid is the cheaper check to return early. The error is not something for users anyway. More like an assert().

Reply:
But the error is reversed: if there is a luks2 and there is no password then something did goes wrong (pbly the user executed jeosfb manually or `touch /var/lib/YaST2/reconfig_system`). The error should be that the keyid is missing and that the enrollment can fail.

The current check is: if there is keyid and no luks2 then I assume that there is a recovery key, but technically this is not an error